domain/domain.go

```diff
@@ -1809,3 +1819,57 @@ var (
 	ErrInfoSchemaChanged = dbterror.ClassDomain.NewStdErr(errno.ErrInfoSchemaChanged,
 		mysql.Message(errno.MySQLErrName[errno.ErrInfoSchemaChanged].Raw+". "+kv.TxnRetryableMark, nil))
 )
+
+// SysProcesses holds the sys processes infos
+type SysProcesses struct {
+	mu      *sync.RWMutex
+	procMap map[uint64]sessionctx.Context
+}
+
+func newSysProcTracker(s SysProcesses) sessionctx.SysProcTracker {
+	tracker := sessionctx.SysProcTracker{}
+	tracker.TrackFunc = func(id uint64, proc sessionctx.Context) error {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		if oldProc, ok := s.procMap[id]; ok && oldProc != proc {
+			return errors.Errorf("The ID is in use: %v", id)
+		}
+		s.procMap[id] = proc
+		proc.GetSessionVars().ConnectionID = id
+		atomic.StoreUint32(&proc.GetSessionVars().Killed, 0)
+		return nil
+	}
+	tracker.UnTrackFunc = func(id uint64) {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		if proc, ok := s.procMap[id]; ok {
+			delete(s.procMap, id)
+			proc.GetSessionVars().ConnectionID = 0
+			atomic.StoreUint32(&proc.GetSessionVars().Killed, 0)
+		}
+	}
+	return tracker
+}
+
+// GetSysProcessList gets list of system ProcessInfo
+func (do *Domain) GetSysProcessList() map[uint64]*util.ProcessInfo {
```

Comment:
I would like to see that `GetSysProcessList` and `KillSysProcess` are methods of `SysProcTracker `